transports/jms/src/main/java/org/mule/transport/jms/JmsMessageDispatcher.java

```diff
@@ -186,6 +186,8 @@ private MuleMessage dispatchMessage(MuleEvent event, boolean doSend) throws Exce
 
             if (useReplyToDestination && replyTo != null)
             {
+                final int timeout = endpoint.getResponseTimeout();
```

Comment:
Should not check if the endpoint timeout is set and if not, then use the event timeout?